network/messageFilter.go

```diff
@@ -48,12 +48,7 @@ func makeMessageFilter(bucketsCount, maxBucketSize int) *messageFilter {
 // CheckMessage checks if the given tag/msg already in the collection, and return true if it was there before the call.
 // Prepends our own random secret to the message to make it hard to abuse hash collisions.
 func (f *messageFilter) CheckIncomingMessage(tag protocol.Tag, msg []byte, add bool, promote bool) bool {
-	hasher := crypto.NewHash()
-	hasher.Write(f.nonce[:])
-	hasher.Write([]byte(tag))
-	hasher.Write(msg)
-	var digest crypto.Digest
-	hasher.Sum(digest[:0])
+	digest := crypto.Hash(append(append(f.nonce[:], []byte(tag)...), msg...))
```

Comment:
I think I deliberately laid it out the other way so as to not do memcpy() operations in append(). I'd be curious to see a benchmark of each coding style.

Reply:
I'll be splitting this PR into two: the one for the hashing and all the rest.